parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -191,6 +230,14 @@ _retrycmd_internal() {
             break # Success, exit loop
         fi
 
+        # TODO mumanski - handle this in different places/modify error messages?
+        # Check if CSE timeout is approaching - exit early to avoid 124 exit code from the global timeout
+        check_cse_timeout $shouldLog
```

Comment:
```suggestion
        if ! check_cse_timeout $shouldLog; then ...
```

Reply:
actually make sure this works with the `return` instead of `exit`, I'm assuming it does tohugh

Reply:
Changed and it is tested by `returns 2 and times out when retrycmd_internal exceeds the CSE timeout` in `cse_retry_helpers_spec.sh`